visual-dotnet/SauceLabs.Visual/VisualClient.cs

```diff
@@ -93,6 +103,70 @@ public VisualClient(WebDriver wd, Region region, string username, string accessK
                 .Build();
         }
 
+        /// <summary>
+        /// <c>FindBuildById</c> returns the build identified by <c>buildId</c>
+        /// </summary>
+        /// <param name="buildId"></param>
+        /// <returns>the matching build</returns>
+        /// <exception cref="VisualClientException">when build is not existing or has an invalid state</exception>
+        private async Task<VisualBuild> FindBuildById(string buildId)
+        {
+            var response = await _api.Build(buildId);
+            if (response.Data?.Result == null)
+            {
+                throw new VisualClientException($@"build {buildId} was not found");
+            }
+
+            var build = response.Data.Result;
+            if (build.Mode != BuildMode.Running)
+            {
+                throw new VisualClientException($"build {build.Id} is not RUNNING");
+            }
+            return new VisualBuild(build.Id, build.Url);
+        }
+
+        /// <summary>
+        /// <c>TryFindBuildByCustomId</c> returns the build identified by <c>customId</c> or null if not found
+        /// </summary>
+        /// <param name="customId"></param>
+        /// <returns>the matching build or null</returns>
+        /// <exception cref="VisualClientException">when build has an invalid state</exception>
+        private async Task<VisualBuild?> TryFindBuildByCustomId(string customId)
+        {
+            var response = await _api.BuildByCustomId(customId);
+            if (response.Data?.Result == null)
+            {
+                return null;
+            }
+
+            var build = response.Data.Result;
+            if (build.Mode != BuildMode.Running)
```

Comment:
Should we raise it or rather extend the VisualBuild class with information about build mode? I think this method shouldn't be responsible for throwing errors such as build is running. I think it belongs to the business logic somewhere else.

Reply:
I can wrap that logic into `VisualBuild`, but it will still be called here.